.github/workflows/ci_workflow.yml

```diff
@@ -0,0 +1,52 @@
+name: Flutter CI
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        uses: ./.github/actions/install_dependencies
+        with:
+          private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      - name: Cache Flutter dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.pub-cache
+            .dart_tool
+          key: ${{ runner.os }}-flutter-${{ hashFiles('**/pubspec.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-flutter-
+
+      - name: Flutter Analyze
+        run: flutter analyze --no-fatal-infos --no-fatal-warnings .
+
+      - name: Run Flutter Tests
+        run: flutter test
+
+      - name: Build Flutter Web
```

Comment:
For pull request validation, do we need building web?

Reply:
We can consider removing it. It was added to align with Bitrise code validation.